lxd/rsync/rsync.go

```diff
@@ -68,6 +68,10 @@ func LocalCopy(source string, dest string, bwlimit string, xattrs bool, rsyncArg
 		return "", err
 	}
 
+	if !shared.IsDir(source) {
```

Comment:
What does this check help with?

Reply:
This check, assuming its needed, would be better done before we make the target dir above.

Reply:
My initial intention was to address the CodeQL alert but that `source` isn't the problem in the end.

Considering that `rsync` tries to `chdir` when the source has a trailing `/`, it might not be needed at all as you suggested.
```
# rsync -a /etc/hostname/ /tmp/foo/
rsync: [sender] change_dir "/etc/hostname" failed: Not a directory (20)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1338) [sender=3.2.7]
```
Happy to drop it if you think the `Source %q is not a directory` error isn't worth it.

Reply:
Yeah let's drop it, it's just an extra syscall not needed